CorePayments/src/main/java/com/paypal/android/corepayments/ResourceLoader.kt

```diff
@@ -0,0 +1,29 @@
+package com.paypal.android.corepayments
+
+import android.content.Context
+import android.content.res.Resources
+import androidx.annotation.RawRes
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+
+class ResourceLoader(context: Context) {
```

Comment:
qq: whats the benefit of putting queries in resource files? Isnt loading a resource a much more expensive operation as if the query lived in the code?

Reply:
That's a good question there's definitely some tradeoffs. I did try putting in a string, but escaping `$`'s in Kotlin felt kind of weird.

It'll be slightly more expensive, but the latency of the web request should be much larger than the latency from reading the file from disk. In total, the user may not notice. We do need access to `Context` as well which kinda sucks, but I think we'll have access to an `applicationContext` reference in most cases.

There are some schema validation plugins too that we can use to make sure the queries are valid. That'll give us some compile time schema validation that isn't available with plain strings.